From: Anupam Kumar <kyteinsky@gmail.com>
Date: Sun, 13 Mar 2022 08:18:11 +0000
Subject: appDisplay: Home and End keys for app grid navigation

Include Home and End keys for consistent behaviour with respect to
overview navigation.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2241>

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/db687493
---
 js/ui/appDisplay.js | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index 4fa2a5f..62d0c9c 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -1669,6 +1669,12 @@ class AppDisplay extends BaseAppView {
         } else if (event.get_key_symbol() === Clutter.KEY_Page_Down) {
             this.goToPage(this._grid.currentPage + 1);
             return Clutter.EVENT_STOP;
+        } else if (event.get_key_symbol() === Clutter.KEY_Home) {
+            this.goToPage(0);
+            return Clutter.EVENT_STOP;
+        } else if (event.get_key_symbol() === Clutter.KEY_End) {
+            this.goToPage(this._grid.nPages - 1);
+            return Clutter.EVENT_STOP;
         }
 
         return Clutter.EVENT_PROPAGATE;
